deepchem/models/optimizers.py

```diff
@@ -550,6 +550,66 @@ def _create_jax_schedule(self):
                                        staircase=self.staircase)
 
 
+class LambdaLRWithWarmup(LearningRateSchedule):
+    """A learning rate scheduler supporting warmup."""
+
+    def __init__(self,
+                 initial_rate: float,
+                 num_warmup_steps: int,
+                 num_training_steps: Optional[int] = None,
+                 warmup_type: str = 'linear'):
+        """
+        Parameters
+        ----------
+        initial_rate: float
+            Initial learning rate
+        num_warmup_steps: int
+            Number of warmup steps
+        num_training_steps: int
+            Number of training steps - required for linear schedule.
+        warmup_type: str, optional. default: linear
+            When `linear`, creates a learning rate schedule that decreases linearly from
+                the initial lr in the optimizer to 0.
+            When `constant`, creates a constant learning rate preceded by a warmup period
+                during which the learning rate increases linearly between 0 and the initial
+                lr set in the optimizer.
+        """
+        assert warmup_type == 'linear' or 'constant', f'Warmup type {warmup_type} is not supported.'
+        self.initial_rate = initial_rate
+        self.num_warmup_steps = num_warmup_steps
+        self.num_training_steps = num_training_steps
+        self.warmup_type = warmup_type
+
+    def _create_pytorch_schedule(self, optimizer):
```

Comment:
Add some comments/docstrings here. Even if it's private, this is complicated enough that would be useful

Reply:
Added docstrings